### PR TITLE
Added flash files to read and write to flash.

### PR DESCRIPTION
**Pull request by gonzagab**
also added some test code to blinky_nucleo64/main.c

### CODE REVIEW

File: firmware/src/peripheral/flash.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#define FLASH_H_
+
+#include "common_header.h"
+
+#define ADDR_FLASH_SECTOR_0     ((uint32_t)0x08000000) /* Base @ of Sector 0, 16 Kbytes */
+
+/**
+ *
+ */
+uint32_t write_flash(uint32_t num_of_bytes, uint8_t* data);
```
Comment (kbumsik) [inline firmware/src/peripheral/flash.h]: In these kinds of functions buffers usually come first. [Like this](http://man7.org/linux/man-pages/man3/memcpy.3.html). Can you swap them?

File: firmware/src/peripheral/flash.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    __HAL_FLASH_DATA_CACHE_RESET();
+    __HAL_FLASH_INSTRUCTION_CACHE_RESET();
+
+    __HAL_FLASH_INSTRUCTION_CACHE_ENABLE();
+    __HAL_FLASH_DATA_CACHE_ENABLE();
+    
+    /* Write Bytes to Flash */
+    for (num_of_bytes; num_of_bytes > 0; num_of_bytes--) {
+        if(HAL_FLASH_Program(FLASH_TYPEPROGRAM_BYTE,  address++, *(data++)) != HAL_OK) {
+            return sector_error;
```
Comment (kbumsik) [inline firmware/src/peripheral/flash.c]: There is no write on `sector_error`. So what it supposed to return?

Comment (gonzagab) [reply firmware/src/peripheral/flash.c]: Thank you i missed that

File: firmware/src/peripheral/flash.h (hunk 1 above)
Comment (gonzagab) [reply firmware/src/peripheral/flash.h]: lolol. got it

**Pull request merged**